File: ui/valset-standardizer-frontend/packages/src/api/parseQueue.ts
import { customInstance } from "./mutator";

export type ParseQueueStatus =
  | "PENDING"
  | "PARSING"
  | "PARSED"
  | "FAILED";

export type ParseQueueTriggerMode = "AUTO" | "MANUAL";

export type ParseQueueViewDTO = {
  queueId?: string;
  businessKey?: string;
  transferId?: string;
  originalName?: string;
  sourceId?: string;
  sourceType?: string;
  sourceCode?: string;
  routeId?: string;
  deliveryId?: string;
  tagId?: string;
  tagCode?: string;
  tagName?: string;
  fileStatus?: string;
  deliveryStatus?: string;
  parseStatus?: ParseQueueStatus;
  triggerMode?: ParseQueueTriggerMode;
  retryCount?: number;
  subscribedBy?: string;
  subscribedAt?: string;
  parsedAt?: string;
  lastErrorMessage?: string;
  objectSnapshotJson?: unknown;
  deliverySnapshotJson?: unknown;
  parseRequestJson?: unknown;
  parseResultJson?: unknown;
  createdAt?: string;
  updatedAt?: string;
};

export type PageResultParseQueueViewDTO = {
  data?: ParseQueueViewDTO[];
  totalCount?: number;
  pageIndex?: number;
  pageSize?: number;
};

export type MultiResultParseQueueViewDTO = {
  data?: ParseQueueViewDTO[];
};

export type SingleResultParseQueueViewDTO = {
  data?: ParseQueueViewDTO | null;
};

export type ParseQueueGenerateCommand = {
  transferId?: string;
  businessKey?: string;
  sourceId?: string;
  routeId?: string;
  tagCode?: string;
  forceRebuild?: boolean;
};

export type ParseQueueBackfillCommand = {
  transferId?: string;
  sourceId?: string;
  sourceCode?: string;
  routeId?: string;
  tagCode?: string;
  status?: string;
  deliveryStatus?: string;
  parseStatus?: string;
  forceRebuild?: boolean;
  dryRun?: boolean;
};

export type ParseQueueSubscribeCommand = {
  subscribedBy?: string;
};

export type ParseQueueCompleteCommand = {
  parseResultJson?: unknown;
};

export type ParseQueueFailCommand = {
  errorMessage?: string;
};

export type ParseQueueRetryCommand = {
  forceRebuild?: boolean;
};

export type ParseQueueQueryParams = {
  transferId?: string;
  businessKey?: string;
  sourceCode?: string;
  routeId?: string;
  tagCode?: string;
  fileStatus?: string;
  deliveryStatus?: string;
  parseStatus?: string;
  triggerMode?: string;
  pageIndex?: number;
  pageSize?: number;
};

export const listParseQueues = (
  params?: ParseQueueQueryParams,
) => {
  return customInstance<PageResultParseQueueViewDTO>({
    url: `/transfer-parse-queues`,
    method: "GET",
    params,
  });
};

export const getParseQueue = (queueId: string) => {
  return customInstance<SingleResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/${queueId}`,
    method: "GET",
  });
};

export const generateParseQueue = (
  command: ParseQueueGenerateCommand,
) => {
  return customInstance<SingleResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/generate`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: command,
  });
};

export const backfillParseQueue = (
  command: ParseQueueBackfillCommand,
) => {
  return customInstance<MultiResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/backfill`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: command,
  });
};

export const subscribeParseQueue = (
  queueId: string,
  command: ParseQueueSubscribeCommand,
) => {
  return customInstance<SingleResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/${queueId}/subscribe`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: command,
  });
};

export const completeParseQueue = (
  queueId: string,
  command: ParseQueueCompleteCommand,
) => {
  return customInstance<SingleResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/${queueId}/complete`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: command,
  });
};

export const failParseQueue = (
  queueId: string,
  command: ParseQueueFailCommand,
) => {
  return customInstance<SingleResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/${queueId}/fail`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: command,
  });
};

export const retryParseQueue = (
  queueId: string,
  command: ParseQueueRetryCommand,
) => {
  return customInstance<SingleResultParseQueueViewDTO>({
    url: `/transfer-parse-queues/${queueId}/retry`,
    method: "POST",
    headers: { "Content-Type": "application/json" },
    data: command,
  });
};

File: valset-standardizer-tools/parser/src/main/java/com/yss/valset/parser/domain/model/ParseQueue.java
package com.yss.valset.parser.domain.model;

import java.time.Instant;
import lombok.Value;

/**
 * 待解析任务。
 */
@Value
public class ParseQueue {

    String queueId;
    String businessKey;
    String transferId;
    String originalName;
    String sourceId;
    String sourceType;
    String sourceCode;
    String routeId;
    String deliveryId;
    String tagId;
    String tagCode;
    String tagName;
    String fileStatus;
    String deliveryStatus;
    ParseStatus parseStatus;
    ParseTriggerMode triggerMode;
    Integer retryCount;
    String subscribedBy;
    Instant subscribedAt;
    Instant parsedAt;
    String lastErrorMessage;
    String objectSnapshotJson;
    String deliverySnapshotJson;
    String parseRequestJson;
    String parseResultJson;
    Instant createdAt;
    Instant updatedAt;

    public String queueId() { return queueId; }
    public String businessKey() { return businessKey; }
    public String transferId() { return transferId; }
    public String originalName() { return originalName; }
    public String sourceId() { return sourceId; }
    public String sourceType() { return sourceType; }
    public String sourceCode() { return sourceCode; }
    public String routeId() { return routeId; }
    public String deliveryId() { return deliveryId; }
    public String tagId() { return tagId; }
    public String tagCode() { return tagCode; }
    public String tagName() { return tagName; }
    public String fileStatus() { return fileStatus; }
    public String deliveryStatus() { return deliveryStatus; }
    public ParseStatus parseStatus() { return parseStatus; }
    public ParseTriggerMode triggerMode() { return triggerMode; }
    public Integer retryCount() { return retryCount; }
    public String subscribedBy() { return subscribedBy; }
    public Instant subscribedAt() { return subscribedAt; }
    public Instant parsedAt() { return parsedAt; }
    public String lastErrorMessage() { return lastErrorMessage; }
    public String objectSnapshotJson() { return objectSnapshotJson; }
    public String deliverySnapshotJson() { return deliverySnapshotJson; }
    public String parseRequestJson() { return parseRequestJson; }
    public String parseResultJson() { return parseResultJson; }
    public Instant createdAt() { return createdAt; }
    public Instant updatedAt() { return updatedAt; }
}
